ext/GraphvizVisualization.jl

```diff
@@ -0,0 +1,667 @@
+module GraphvizVisualization
+
+export generate_dot, show_gv, dot_string_to_pdf, SimpleIteration, BFSTraversal
+
+using GraphPPL, MetaGraphsNext, GraphViz
+using GraphPPL.MetaGraphsNext
+import MetaGraphsNext: nv
+
+"""
+This abstract type represents a node traversal strategy for use with the `generate_dot` function.
+
+This abstract type is used to define various strategies for traversing nodes in the graph when generating a DOT representation. 
+Each concrete subtype specifies a different traversal approach, which is selected by Julia's multiple dispatch system 
+when calling `generate_dot`.
+
+Concrete subtypes:
+- `SimpleIteration`: Represents a simple iteration of the graph's vertex/node set.
+- `BFSTraversal`: Represents a breadth-first search traversal strategy, from the initially created node.
+
+These types afford a trade-off between a relatively fast and a relatively 'principled' iteration strategy (respectfully).
+"""
+abstract type TraversalStrategy end
+struct SimpleIteration <: TraversalStrategy end
+struct BFSTraversal <: TraversalStrategy end
```

Comment:
BTW I really like this strategy type dispatch so I really want to keep this, great work

Reply:
I like it too, but since we cannot access them from the extension maybe just use `Symbol`s instead, e.g. `:simple_iteration` and `:bfs_traversal` (which can be converted to their corresponding structures if necessary and yes it will be type-unstable but this is not performance critical code and the type-instability will be negligible )

Reply:
e.g. something similar to
```julia
convert_symbol_strategy_to_struct(::Val{:simple_iteration}) = SimpleIteration()
convert_symbol_strategy_to_struct(::Val{:bfs_traversal}) = BFSTraversal()

function plot(; strategy = :simple_iteration)
    strategy_struct = convert_symbol_strategy_to_struct(Val(strategy))
    ...
end
```